cluster-autoscaler/cloudprovider/gce/gce_manager.go

```diff
@@ -699,18 +709,60 @@ func (m *gceManagerImpl) getTemplates() *templateBuilder {
 }
 
 func (m *gceManagerImpl) Refresh() error {
+	glog.Warningf("refresh debug 1")
 	if m.mode == ModeGCE {
 		return nil
 	}
 	if m.lastRefresh.Add(refreshInterval).Before(time.Now()) {
+		glog.Warningf("refresh debug 2")
 		err := m.fetchAllNodePools()
+		if err != nil {
+			return err
+		}
+
+		err = m.fetchResourceLimiter()
+		if err != nil {
+			return err
+		}
+
 		m.lastRefresh = time.Now()
-		glog.V(2).Infof("Refreshed NodePools list, next refresh after %v", m.lastRefresh.Add(refreshInterval))
-		return err
+		glog.V(2).Infof("Refreshed NodePools list and resource limits, next refresh after %v", m.lastRefresh.Add(refreshInterval))
+		return nil
 	}
 	return nil
 }
 
+// GetMigNodes returns resource limiter.
+func (m *gceManagerImpl) fetchResourceLimiter() error {
+	if m.mode == ModeGKENAP {
+		cluster, err := m.gkeAlphaService.Projects.Zones.Clusters.Get(m.projectId, m.zone, m.clusterName).Do()
+		if err != nil {
+			return nil
+		}
+
+		minLimits := make(map[string]int64)
+		maxLimits := make(map[string]int64)
+		for _, limit := range cluster.Autoscaling.ResourceLimits {
+			minLimits[limit.Name] = limit.Minimum
+			maxLimits[limit.Name] = limit.Maximum
```

Comment:
Maybe we should check limit.Name against a list of known limits? We could filter those out and log some warnings (though I'd rather not return error to keep forward compatibility). WDYT @krzysztof-jastrzebski?

Reply:
I don't think we should maintain list of resource names in code. In the future we might want  introduce limits without changing code (like number of accelerators of type X that are new).

Reply:
We don't currently have the code to respect limits on GPUs, so we'll have to change it anyway. In general, I think it makes sense to log that we aren't going to respect a given limit because CA doesn't understand it. This of course doesn't mean it has to be a fixed list (for example, we may identify accelerator types by prefix in the future.)

Reply:
Added.